### PR TITLE
[PfemFluid] new remeshing and material property settings for two-fluid analysis

### PR DESCRIPTION
**Pull request by AFranci**
Property ID assignation and remeshing opertations (adding/removing nodes) are now performed to maintain as much as possible the orginal volume of each material in a multi-fluid analysis. These changes improve sensibly the previous implementation.

### CODE REVIEW

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp [hunk 1]
```diff
@@ -1761,8 +1787,17 @@ namespace Kratos
 						Edges[i] = 0;
 					}
 				}
+				else if (rigidNodes == 0)
+				{
+					unsigned int propertyIdFirstNode = Element[FirstEdgeNode[i]].FastGetSolutionStepValue(PROPERTY_ID);
+					unsigned int propertyIdSecondNode = Element[SecondEdgeNode[i]].FastGetSolutionStepValue(PROPERTY_ID);
+					if (propertyIdFirstNode != propertyIdSecondNode)
+					{
+						penalization = 0.8;
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp:1796]: I like magic numbers haha

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp [hunk 2]
```diff
@@ -2010,22 +2045,30 @@ namespace Kratos
...
-					{
-						TakeMaterialPropertiesFromNotRigidNode(pnode, SlaveNode2);
-					}
+
+					TakeMaterialPropertiesFromNotRigidNode(pnode, SlaveNode1);
+					// unsigned int propertyIdFirstNode = SlaveNode1->FastGetSolutionStepValue(PROPERTY_ID);
+					// unsigned int propertyIdSecondNode = SlaveNode2->FastGetSolutionStepValue(PROPERTY_ID);
+					// if (propertyIdFirstNode != propertyIdSecondNode && SlaveNode1->IsNot(RIGID) && SlaveNode2->IsNot(RIGID))
+					// {
+					// 	std::cout << "I SHOULD NOT ENTER HEEEEEEEEEEEEERE " << propertyIdFirstNode << " VS " << propertyIdSecondNode << std::endl;
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp]: we need this commented code?

File: applications/PfemFluidDynamicsApplication/custom_processes/remove_mesh_nodes_for_fluids_process.hpp [hunk 3]
```diff
@@ -622,7 +630,7 @@ namespace Kratos
 
 								if ((*nn)->Is(BOUNDARY) && !nn_on_contact_tip && neighbour_distances[k] < size_for_distance_boundary && neighbour_distances[k] > 0.0)
 								{
-									//KRATOS_WATCH( neighbours_distances[k] )
+									// KRATOS_WATCH( neighbours_distances[k] )
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/remove_mesh_nodes_for_fluids_process.hpp]: why not removing prints?

File: applications/PfemFluidDynamicsApplication/custom_processes/set_material_properties_from_fluid_to_rigid_nodes_process.hpp [hunk 4]
```diff
@@ -133,19 +133,19 @@ namespace Kratos
           bulk_modulus = elemProperties[BULK_MODULUS];
           viscosity = elemProperties[DYNAMIC_VISCOSITY];
 
-          if (elemProperties.Has(YIELD_SHEAR)) //Bingham model
+          if (elemProperties.Has(YIELD_SHEAR)) // Bingham model
           {
             flow_index = elemProperties[FLOW_INDEX];
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/set_material_properties_from_fluid_to_rigid_nodes_process.hpp:138]: you checked Yield but this variable has not been checked, can be dangeorus

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp (hunk 2 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp]: I should have removed it with the last committ

File: applications/PfemFluidDynamicsApplication/custom_processes/remove_mesh_nodes_for_fluids_process.hpp (hunk 3 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_processes/remove_mesh_nodes_for_fluids_process.hpp]: ok, removed with the new commit

File: applications/PfemFluidDynamicsApplication/custom_processes/set_material_properties_from_fluid_to_rigid_nodes_process.hpp (hunk 4 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_processes/set_material_properties_from_fluid_to_rigid_nodes_process.hpp:138]: The rest of variables are present if there is Yield_shear. They are required by the Bingham model

**Comment by AlejandroCornejo** [comment]
Do you think that we should add a test?

**Comment by AFranci** [comment]
> Do you think that we should add a test?

done!

**Review by AlejandroCornejo — APPROVED** [review]
Hell yeah!

**Pull request merged**